de-lib/src/main/java/org/iplantc/de/client/models/UserInfo.java

```diff
@@ -78,8 +80,41 @@ public void init(String userInfoJson) {
         preferences = userBootstrap.getPreferences();
         session = userBootstrap.getSession();
         appsInfo = userBootstrap.getAppsInfo();
-        workspace = appsInfo.getWorkspace();
-        systemIds = appsInfo.getSystemsIds();
+        if (appsInfo != null) {
+            workspace = appsInfo.getWorkspace();
+            systemIds = appsInfo.getSystemsIds();
+        }
+
+        checkForErrors();
+    }
+
+    private void checkForErrors() {
+        if (userProfile == null || hasUserProfileError()) {
+            logBootStrapError(UserBootstrap.USER_INFO_KEY, null);
+        }
+        if (dataInfo == null || hasDataInfoError()) {
+            logBootStrapError(UserBootstrap.DATA_INFO_KEY, dataInfo);
+        }
+        if (preferences == null || hasPreferencesError()) {
+            logBootStrapError(UserBootstrap.PREFERENCES_KEY, preferences);
+        }
+        if (session == null || hasSessionError()) {
+            logBootStrapError(UserBootstrap.SESSION_KEY, session);
+        }
+        if (appsInfo == null || hasAppsInfoError()) {
+            logBootStrapError(UserBootstrap.APPS_INFO_KEY, appsInfo);
+        }
+    }
+
+    private void logBootStrapError(String key, HasSplittableError obj) {
+        String errorMsg;
+        errorMsg = "Bootstrap error within '" + key + "' key: ";
+        if (obj != null) {
+            errorMsg += obj.getError().getPayload();
+        } else {
+            errorMsg += "null value or key parsing error";
+        }
+        LOG.severe(errorMsg);
```

Comment:
Do we have do any kind of set-up for server side logging? I am referring to http://www.gwtproject.org/doc/latest/DevGuideLogging.html#Client_vs_Server_side_Logging